sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/HadoopFsRelation.scala

```diff
@@ -82,7 +84,15 @@ case class HadoopFsRelation(
     }
   }
 
-  override def sizeInBytes: Long = location.sizeInBytes
+  override def sizeInBytes: Long = {
+    val size = location.sizeInBytes * hadoopFSSizeFactor
```

Comment:
Overflow?

Reply:
this should be handled by `size` > Long.MaxValue, the double value is overflowed only when the result value is Double.PositivyInfinity which would be capped as Long.MaxValue

Reply:
```
scala> val size: Long = 0x7fffffffffffffffL
size: Long = 9223372036854775807

scala> val res = size * 10
res: Long = -10
```

Reply:
nvm. hadoopFSSizeFactor is a double